### ISSUE TITLE
glide panics trying to generate lockfile with `--no-recursive`

### ISSUE THREAD

**Issue by kshlm** [Masterminds/glide#165]
```shell
$ glide --version
glide version 0.8.0
```

I'm using glide to manage dependencies for a project of mine. The recursive update mechanism enabled in 0.8.0, causes glide to download a huge number of unneeded packages. I would like to add only the required packages to glide.yaml and use `--no-recursive` to download just them. But using `--no-recursive` causes update to fail with a panic, `panic: interface conversion: interface {} is nil, not *cfg.Lockfile` (the full backtrace is at the end).

I use `github.com/docker/libkv` which has a sort of plugin architecture, which allows me to choose from multiple backends (etcd, consul, bolt and zookeeper). As I intend to use just one of the backends, the others don't matter to me. Glide pulls in all the dependencies for libkv, including etcd, consul and their large list of dependencies. I want to use `--no-recursive` to avoid this.

```
panic: interface conversion: interface {} is nil, not *cfg.Lockfile

goroutine 1 [running]:
github.com/Masterminds/glide/cmd.WriteLock(0x7f25c17a86c8, 0xc820111410, 0xc820032318, 0x0, 0x0, 0x0, 0x0)
        /Users/mfarina/Code/go/src/github.com/Masterminds/glide/cmd/yaml.go:108 +0x456
github.com/Masterminds/glide/vendor/github.com/Masterminds/cookoo.(*Router).doCommand(0xc82013cea0, 0xc820160810, 0x7f25c17a86c8, 0xc820111410, 0x0, 0x0, 0x0, 0x0)
        /Users/mfarina/Code/go/src/github.com/Masterminds/glide/vendor/github.com/Masterminds/cookoo/router.go:209 +0x86
github.com/Masterminds/glide/vendor/github.com/Masterminds/cookoo.(*Router).runRoute(0xc82013cea0, 0x9508b8, 0x6, 0x7f25c17a86c8, 0xc820111410, 0x950800, 0x0, 0x0)
        /Users/mfarina/Code/go/src/github.com/Masterminds/glide/vendor/github.com/Masterminds/cookoo/router.go:164 +0x4c9
github.com/Masterminds/glide/vendor/github.com/Masterminds/cookoo.(*Router).HandleRequest(0xc82013cea0, 0x9508b8, 0x6, 0x7f25c17a86c8, 0xc820111410, 0xc8200eb000, 0x0, 0x0)
        /Users/mfarina/Code/go/src/github.com/Masterminds/glide/vendor/github.com/Masterminds/cookoo/router.go:131 +0x2fc
main.setupHandler(0xc82016c000, 0x9508b8, 0x6, 0x7f25c17a86c8, 0xc820111410, 0xc82013cea0)
        /Users/mfarina/Code/go/src/github.com/Masterminds/glide/glide.go:514 +0x48c
main.commands.func10(0xc82016c000)
        /Users/mfarina/Code/go/src/github.com/Masterminds/glide/glide.go:466 +0x89f
github.com/Masterminds/glide/vendor/github.com/codegangsta/cli.Command.Run(0x9508b8, 0x6, 0x9508b0, 0x2, 0x0, 0x0, 0x0, 0x9aaf40, 0x1f, 0xa55380, ...)
        /Users/mfarina/Code/go/src/github.com/Masterminds/glide/vendor/github.com/codegangsta/cli/command.go:137 +0x107e
github.com/Masterminds/glide/vendor/github.com/codegangsta/cli.(*App).Run(0xc820019d40, 0xc82000a390, 0x3, 0x3, 0x0, 0x0)
        /Users/mfarina/Code/go/src/github.com/Masterminds/glide/vendor/github.com/codegangsta/cli/app.go:175 +0xfff
main.main()
        /Users/mfarina/Code/go/src/github.com/Masterminds/glide/glide.go:119 +0x73e

goroutine 21 [select]:
net/http.(*persistConn).writeLoop(0xc8200fac60)
        /usr/local/Cellar/go/1.5.2/libexec/src/net/http/transport.go:1009 +0x40c
created by net/http.(*Transport).dialConn
        /usr/local/Cellar/go/1.5.2/libexec/src/net/http/transport.go:686 +0xc9d

goroutine 18 [IO wait]:
net.runtime_pollWait(0x7f25c17a9c58, 0x72, 0xc820016210)
        /usr/local/Cellar/go/1.5.2/libexec/src/runtime/netpoll.go:157 +0x60
net.(*pollDesc).Wait(0xc82015cc30, 0x72, 0x0, 0x0)
        /usr/local/Cellar/go/1.5.2/libexec/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc82015cc30, 0x0, 0x0)
        /usr/local/Cellar/go/1.5.2/libexec/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).Read(0xc82015cbd0, 0xc8201c0000, 0x2000, 0x2000, 0x0, 0x7f25c17a4050, 0xc820016210)
        /usr/local/Cellar/go/1.5.2/libexec/src/net/fd_unix.go:232 +0x23a
net.(*conn).Read(0xc8201aa008, 0xc8201c0000, 0x2000, 0x2000, 0x0, 0x0, 0x0)
        /usr/local/Cellar/go/1.5.2/libexec/src/net/net.go:172 +0xe4
crypto/tls.(*block).readFromUntil(0xc8201a6240, 0x7f25c17680a8, 0xc8201aa008, 0x5, 0x0, 0x0)
        /usr/local/Cellar/go/1.5.2/libexec/src/crypto/tls/conn.go:455 +0xcc
crypto/tls.(*Conn).readRecord(0xc8201b8000, 0xa49a17, 0x0, 0x0)
        /usr/local/Cellar/go/1.5.2/libexec/src/crypto/tls/conn.go:540 +0x2d1
crypto/tls.(*Conn).Read(0xc8201b8000, 0xc820222000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
        /usr/local/Cellar/go/1.5.2/libexec/src/crypto/tls/conn.go:901 +0x167
net/http.noteEOFReader.Read(0x7f25c1770a90, 0xc8201b8000, 0xc8200fac08, 0xc820222000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
        /usr/local/Cellar/go/1.5.2/libexec/src/net/http/transport.go:1370 +0x67
net/http.(*noteEOFReader).Read(0xc82049a560, 0xc820222000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
        <autogenerated>:126 +0xd0
bufio.(*Reader).fill(0xc820429020)
        /usr/local/Cellar/go/1.5.2/libexec/src/bufio/bufio.go:97 +0x1e9
bufio.(*Reader).Peek(0xc820429020, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0)
        /usr/local/Cellar/go/1.5.2/libexec/src/bufio/bufio.go:132 +0xcc
net/http.(*persistConn).readLoop(0xc8200fabb0)
        /usr/local/Cellar/go/1.5.2/libexec/src/net/http/transport.go:876 +0xf7
created by net/http.(*Transport).dialConn
        /usr/local/Cellar/go/1.5.2/libexec/src/net/http/transport.go:685 +0xc78

goroutine 19 [select]:
net/http.(*persistConn).writeLoop(0xc8200fabb0)
        /usr/local/Cellar/go/1.5.2/libexec/src/net/http/transport.go:1009 +0x40c
created by net/http.(*Transport).dialConn
        /usr/local/Cellar/go/1.5.2/libexec/src/net/http/transport.go:686 +0xc9d

goroutine 20 [IO wait]:
net.runtime_pollWait(0x7f25c17a9d18, 0x72, 0xc820016210)
        /usr/local/Cellar/go/1.5.2/libexec/src/runtime/netpoll.go:157 +0x60
net.(*pollDesc).Wait(0xc8204cc140, 0x72, 0x0, 0x0)
        /usr/local/Cellar/go/1.5.2/libexec/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc8204cc140, 0x0, 0x0)
        /usr/local/Cellar/go/1.5.2/libexec/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).Read(0xc8204cc0e0, 0xc82014b400, 0x400, 0x400, 0x0, 0x7f25c17a4050, 0xc820016210)
        /usr/local/Cellar/go/1.5.2/libexec/src/net/fd_unix.go:232 +0x23a
net.(*conn).Read(0xc820032468, 0xc82014b400, 0x400, 0x400, 0x0, 0x0, 0x0)
        /usr/local/Cellar/go/1.5.2/libexec/src/net/net.go:172 +0xe4
crypto/tls.(*block).readFromUntil(0xc82019cab0, 0x7f25c17680a8, 0xc820032468, 0x5, 0x0, 0x0)
        /usr/local/Cellar/go/1.5.2/libexec/src/crypto/tls/conn.go:455 +0xcc
crypto/tls.(*Conn).readRecord(0xc8200cfb80, 0xa49a17, 0x0, 0x0)
        /usr/local/Cellar/go/1.5.2/libexec/src/crypto/tls/conn.go:540 +0x2d1
crypto/tls.(*Conn).Read(0xc8200cfb80, 0xc8204f3000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
        /usr/local/Cellar/go/1.5.2/libexec/src/crypto/tls/conn.go:901 +0x167
net/http.noteEOFReader.Read(0x7f25c1770a90, 0xc8200cfb80, 0xc8200facb8, 0xc8204f3000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
        /usr/local/Cellar/go/1.5.2/libexec/src/net/http/transport.go:1370 +0x67
net/http.(*noteEOFReader).Read(0xc820511060, 0xc8204f3000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
        <autogenerated>:126 +0xd0
bufio.(*Reader).fill(0xc820429ec0)
        /usr/local/Cellar/go/1.5.2/libexec/src/bufio/bufio.go:97 +0x1e9
bufio.(*Reader).Peek(0xc820429ec0, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0)
        /usr/local/Cellar/go/1.5.2/libexec/src/bufio/bufio.go:132 +0xcc
net/http.(*persistConn).readLoop(0xc8200fac60)
        /usr/local/Cellar/go/1.5.2/libexec/src/net/http/transport.go:876 +0xf7
created by net/http.(*Transport).dialConn
        /usr/local/Cellar/go/1.5.2/libexec/src/net/http/transport.go:685 +0xc78

goroutine 86 [select]:
net/http.(*persistConn).writeLoop(0xc82035c000)
        /usr/local/Cellar/go/1.5.2/libexec/src/net/http/transport.go:1009 +0x40c
created by net/http.(*Transport).dialConn
        /usr/local/Cellar/go/1.5.2/libexec/src/net/http/transport.go:686 +0xc9d

goroutine 85 [IO wait]:
net.runtime_pollWait(0x7f25c17a9ad8, 0x72, 0xc820016210)
        /usr/local/Cellar/go/1.5.2/libexec/src/runtime/netpoll.go:157 +0x60
net.(*pollDesc).Wait(0xc8204cda30, 0x72, 0x0, 0x0)
        /usr/local/Cellar/go/1.5.2/libexec/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc8204cda30, 0x0, 0x0)
        /usr/local/Cellar/go/1.5.2/libexec/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).Read(0xc8204cd9d0, 0xc820324400, 0x400, 0x400, 0x0, 0x7f25c17a4050, 0xc820016210)
        /usr/local/Cellar/go/1.5.2/libexec/src/net/fd_unix.go:232 +0x23a
net.(*conn).Read(0xc8204ce2e0, 0xc820324400, 0x400, 0x400, 0x0, 0x0, 0x0)
        /usr/local/Cellar/go/1.5.2/libexec/src/net/net.go:172 +0xe4
crypto/tls.(*block).readFromUntil(0xc8202f6ae0, 0x7f25c17680a8, 0xc8204ce2e0, 0x5, 0x0, 0x0)
        /usr/local/Cellar/go/1.5.2/libexec/src/crypto/tls/conn.go:455 +0xcc
crypto/tls.(*Conn).readRecord(0xc8200c0000, 0xa49a17, 0x0, 0x0)
        /usr/local/Cellar/go/1.5.2/libexec/src/crypto/tls/conn.go:540 +0x2d1
crypto/tls.(*Conn).Read(0xc8200c0000, 0xc8202fa000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
        /usr/local/Cellar/go/1.5.2/libexec/src/crypto/tls/conn.go:901 +0x167
net/http.noteEOFReader.Read(0x7f25c1770a90, 0xc8200c0000, 0xc82035c058, 0xc8202fa000, 0x1000, 0x1000, 0xc8202f3d40, 0x0, 0x0)
        /usr/local/Cellar/go/1.5.2/libexec/src/net/http/transport.go:1370 +0x67
net/http.(*noteEOFReader).Read(0xc8203fe280, 0xc8202fa000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
        <autogenerated>:126 +0xd0
bufio.(*Reader).fill(0xc8204d2d80)
        /usr/local/Cellar/go/1.5.2/libexec/src/bufio/bufio.go:97 +0x1e9
bufio.(*Reader).Peek(0xc8204d2d80, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0)
        /usr/local/Cellar/go/1.5.2/libexec/src/bufio/bufio.go:132 +0xcc
net/http.(*persistConn).readLoop(0xc82035c000)
        /usr/local/Cellar/go/1.5.2/libexec/src/net/http/transport.go:876 +0xf7
created by net/http.(*Transport).dialConn
        /usr/local/Cellar/go/1.5.2/libexec/src/net/http/transport.go:685 +0xc78

```

**Comment by mattfarina** [comment]
@kshlm While I take a look into this problem you may want to try to `ignore` property in the `glide.yaml` file. It is a list of packages to ignore fetching and is respected through the dependency tree.

For example,
```yaml
ignore:
- github.com/coreos/etcd
```

I'll try and reproduce the problem with the panic, which shouldn't happen.

**Comment by mattfarina** [comment]
I can reproduce this and I'll ship a fix in 0.8.1 which will be out in the next day or so.

**Comment by kshlm** [comment]
@mattfarina Looking forward to the fix. Thank you.
